### PR TITLE
[ingester] support extract an integer from traceId as an additional i…

### PR DESCRIPTION
**Pull request by lzf575**
…ndex to speed up traceId queries

### This PR is for:

- Server

### CODE REVIEW

File: server/server.yaml [hunk 1]
```diff
@@ -18,6 +18,14 @@ log-level: info
 #  block-rate: 5 # valid when ProfileTypes contains 'block_count' or 'block_duration'
 #  log-enabled: true # whether record profile debug logs
 
+# extract an integer (generally used timestamp) from traceId as an additional index to speed up traceId queries
+#trace-id-with-index:
+#  enabled: false
+#  position: subffix  # prefix/suffix
```
Comment (sharang) [inline server/server.yaml:24]: subffix 拼错了

Comment (lzf575) [reply server/server.yaml]: fix

File: server/server.yaml [hunk 2]
```diff
@@ -18,6 +18,16 @@ log-level: info
 #  block-rate: 5 # valid when ProfileTypes contains 'block_count' or 'block_duration'
 #  log-enabled: true # whether record profile debug logs
 
+# extract an integer (generally used timestamp) from traceId as an additional index to speed up traceId queries.
+#trace-id-with-index:
+#  disabled: false
+#  type: hash           # hash/position
```
Comment (sharang) [inline server/server.yaml:24]: hash, timestamp

File: server/server.yaml [hunk 3]
```diff
@@ -18,6 +18,16 @@ log-level: info
...
+#  #eg. traceId: 'abcdef1234',  if want to match index '1234', `start` can be set as 6 or -4 , and `length` must set: 4
+#  index-position:      # it is valid when 'type' is 'position'
+#    start: 0           # >= 0 means counting from the beginning, < 0 means counting from the end.
+#    length: 13         # eg. 1701419226688
+#    format: dec        # hex/dec
```
Comment (sharang) [inline server/server.yaml:29]: dec 改成 decimal?

单看 dec 有歧义，比如「减少」也叫 dec

Comment (lzf575) [reply server/server.yaml]: ok

File: server/server.yaml (hunk 2 above)
Comment (lzf575) [reply server/server.yaml:24]: 还是保留吧，单独的开关控制比较明了

**Review by sharang — APPROVED** [review]

**Pull request merged**